buildkite/resource_pipeline_team.go

```diff
@@ -0,0 +1,250 @@
+}
+
+func newPipelineTeamResource() resource.Resource {
+	return &pipelineTeamResource{}
+}
+
+func (pipelineTeamResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pipeline_team"
+}
+
+func (tp *pipelineTeamResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	tp.client = req.ProviderData.(*Client)
+}
+
+func (pipelineTeamResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resource_schema.Schema{
+		MarkdownDescription: "A team pipeline resource sets a team's access for the pipeline.",
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"team_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The GraphQL ID of the team.",
+			},
+			"pipeline_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The GraphQL ID of the pipeline.",
+			},
+			"access_level": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The access level for the team. Either READ_ONLY, BUILD_AND_READ or MANAGE_BUILD_AND_READ.",
+				Validators: []validator.String{
+					stringvalidator.OneOf("READ_ONLY", "BUILD_AND_READ", "MANAGE_BUILD_AND_READ"),
+				},
+			},
+		},
+	}
+}
+
+func (tp *pipelineTeamResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state pipelineTeamResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiResponse, err := createTeamPipeline(
+		tp.client.genqlient,
+		state.TeamId.ValueString(),
+		state.PipelineId.ValueString(),
+		PipelineAccessLevels(state.AccessLevel.ValueString()),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create team pipeline",
+			fmt.Sprintf("Unable to create team pipeline: %s", err.Error()),
+		)
+		return
+	}
+
+	// Update state with values from API response/plan
+	state.Id = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Id)
+	state.Uuid = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Uuid)
+	state.PipelineId = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Pipeline.Id)
+	state.TeamId = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Team.Id)
+	state.AccessLevel = types.StringValue(string(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.PipelineAccessLevel))
```

Comment:
I don't think these need to be in here; they're known at run time and so can just be a part of the `resp.State.Set`.

Reply:
Yeah I think you only need to set id and uuid here right? Maybe access level depending on whether its required or not (as above)